elytron/src/test/java/org/wildfly/extension/elytron/RealmsTestCase.java

```diff
@@ -246,6 +254,87 @@ public void testFileSystemRealmEncodingAndCharset() throws Exception {
 
     }
 
+    /**
+     * Test the filesystem realm can handle identities with different types of encrypted passwords and attributes. Also
+     * ensures a pre-assigned SecretKey as well as a newly generated SecretKey work.
+     */
+    @Test
+    public void testFilesystemRealmEncrypted() throws Exception {
+        KernelServices services = super.createKernelServicesBuilder(new TestEnvironment()).setSubsystemXmlResource("realms-test.xml").build();
+        if (!services.isSuccessfulBoot()) {
+            Assert.fail(services.getBootError().toString());
+        }
+
+        ServiceName serviceName = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("FilesystemRealm5");
+        ModifiableSecurityRealm securityRealm = (ModifiableSecurityRealm) services.getContainer().getService(serviceName).getValue();
+        ServiceName serviceName_randomKey = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("FilesystemRealm6");
+        ModifiableSecurityRealm securityRealm_randomKey = (ModifiableSecurityRealm) services.getContainer().getService(serviceName_randomKey).getValue();
+
+        // Test Clear Encrypted Password
+        Assert.assertNotNull(securityRealm);
+        char[] password = "secretPassword".toCharArray();
+        RealmIdentity identity_clear = securityRealm.getRealmIdentity(fromName("plainUser"));
```

Comment:
A minor comment but just noticed this files contains both camelCase variable names and snake_case variable names. We typically use camelCase and use it throughout the file.

Reply:
Fixed